packages/docs/community/release-process/smart-contracts.md

```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
+### Submit Upgrade Proposal
 
-#### Script 2:
+Submit the autogenerated upgrade proposal to the Governance contract for review by voters, outputting a unique identifier.
 
-1. Compiles the contracts in the current branch and checks backwards compatibility with what is currently deployed on the specified network, with the following exceptions:
-   1. If the STORAGE version has changed, does not perform backwards compatibility checks
-   2. If the MAJOR version has changed, checks that the storage layout is backwards compatible, but does not check that the contract ABI is backwards compatible.
-2. For contracts that have changed, confirms that the version number in the current branch is strictly greater than the deployed version number.
-3. For contracts that have not changed, confirms that the version number in the current branch is exactly the same as the deployed version number
+```bash
+celocli governance:propose <...> --jsonTransactions $PROPOSAL
+# save id to $UPGRADE_PROPOSAL_ID
+```
 
-#### Script 3:
+## Verify Release Candidate
 
-1. For contracts that have changed, deploys those contracts to the specified network.
-2. Creates and submits a single governance proposal to upgrade to the newly deployed contracts.
-   1. STORAGE updates are adopted by deploying a new proxy and implementation and updating the Registry contract.
-   2. All other updates are adopted by updating the proxy contract’s implementation pointer.
+Using the following procedure, stakeholders can verify a contract release candidate against the contents of a governance upgrade proposal.
 
-## Verify Release Process
+### Fetch Upgrade Proposal
 
-Using the following script, any stakeholders can verify that the release contracts deployed as a part of the upgrade proposal match the contracts from the release branch exactly. Run this script from the relevant release tag (`celo-core-contracts-v${N}.(baklava | alfajores | mainnet)`) or release branch `release/celo-core-contracts/N`:
+Fetch the upgrade proposal identified by <code>$UPGRADE_PROPOSAL_ID</code> and output the JSON encoded proposal contents to <code>$UPGRADE_PROPOSAL</code>.
 
 ```bash
-# verify bytecode of upgrade deployment
-yarn truffle exec --network development ./scripts/truffle/verify-bytecode.js --build_artifacts build/contracts --proposal ../../proposal.json
+celocli governance:show <...> --proposalID $UPGRADE_PROPOSAL_ID --jsonFile $UPGRADE_PROPOSAL
 ```
 
-This script compiles the contracts at the current branch and confirms that the compiled bytecode matches the bytecode of contracts deployed as a part of the ugrade propsal.
+### Verify Release Candidate
+
+Verify that the <code>$UPGRADE_PROPOSAL</code> proposal performs an upgrade to contract addresses which match compiled bytecode from the release candidate <code>$BRANCH</code> branch exactly. Releases are tagged as `celo-core-contracts-v${N}.(baklava | alfajores | mainnet)`.
```

Comment:
I think specifically $BRANCH in this context should be the audited commit?

Reading this now, I wonder if we should only tag that commit with `celo-core-contracts-v${N}.(baklava | alfajores | mainnet)` once the proposal actually goes through?